pdr_backend/lake/duckdb_data_store.py

```diff
@@ -243,34 +288,79 @@ def query_data(self, query: str) -> Optional[pl.DataFrame]:
             raise e
 
     @enforce_types
-    def move_table_data(self, temp_table, permanent_table):
+    def get_query_drop_records_from_table_by_id(
+        self, drop_table_name: str, ref_table_name: str
+    ):
         """
-        Move the table data from the temporary storage to the permanent storage.
+        Builds the query string to perform the operation and returns it
         @arguments:
-            temp_table - The temporary table object
-            permanent_table_name - The name of the permanent table.
+            drop_table_name - The table to drop records from.
+            ref_table_name - The table to reference for the IDs to drop.
+        @example:
+            get_query_drop_records_from_table_by_id("bronze_pdr_slots", "update_pdr_slots")
+        """
+
+        # Return the query string
+        return f"""
+        DELETE FROM {drop_table_name}
+        WHERE ID IN (
+            SELECT DISTINCT ID 
+            FROM {ref_table_name}
+        );
+        """
+
+    @enforce_types
+    def drop_records_from_table_by_id(self, drop_table_name: str, ref_table_name: str):
```

Comment:
I think this function name is a bit non-descriptive. Maybe something like `drop_from_table_using_reference_table_ids`? Because otherwise I expect one of the parameters to be a list of ids.

Reply:
I have renamed it from `drop_from_table_using_reference_table_ids` -> `drop_common_records_by_id`

However `drop_common_records_by_id` isn't actually being used, as I deprecated it's usage and instead build the query so they whole swap strategy can be executed atomically.

![image](https://github.com/user-attachments/assets/dce8f552-3640-4a70-988d-3cb841a10c11)

So, now we only have: `get_query_drop_common_records_by_id" and "get_query_move_table_data"